File: web_ui/ruby/app/models/person_page_rank.rb
class PersonPageRank < ActiveRecord::Base
  include Models::ColumnMethodsMysql
  self.table_name  = "PersonPageRanks"
  self.primary_key = 'PageID'

  belongs_to :person, primary_key: 'PersonID'
  belongs_to :page,   primary_key: 'PageID'

  validates(:person, :page, presence: true)
  validates(:rank, presence: true)


end

File: WebAPI/CSharp/RSPUserApi/Domain/Entities/PersonPageRank.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    [Table("PersonPageRanks")]
    public class PersonPageRank
    {
        [Key, Column(Order = 1)]
        public int PageID { get; set; }
        [ForeignKey("PageID")]
        public virtual Page Page { get; set; }
        [Key, Column (Order = 2)]
        public int PersonID { get; set; }
        [ForeignKey("PersonID")]
        public virtual Person Person { get; set; }
        [Required]
        public int Rank { get; set; }
    }
}
